### ISSUE TITLE
Update URL for Table

### ISSUE THREAD

**Issue by henry296** [Esri/ago-assistant#104]
Hi have standalone data tables in my web maps because they are related to my spatial features.  How can I update their URL?  When I update the spatial layers URL, the related records link in the pop-up of my web map / app is now broken because the related tables is pointed to a different ArcGIS server REST endpoint than the feature.

The pop-up attributes are pulling from the right source.

**Comment by ecaldwell** [comment]
Are they hosted tables? Do they show up as an option when you choose to "Update the URLs of Services in a Web Map"? If not, you could try editing the JSON of the web map directly and update the URLs there. I suggest you save a copy of the web map first and test the changes against the copy.

**Comment by henry296** [comment]
Evan,

Thanks for the quick response.

The tables are from ArcGIS Server and don’t show up on the Update the URLs of Services in a Web Map. I also noticed they don’t show up in the web map properties in ArcGIS Online.   While the tables are not hosted, the web map is hosted so I can’t update the JSON directly.

The only workaround I have found is to delete the tables are re-add the entire service which includes the feature layer and the related table.


[cid:image001.jpg@01CEB48E.2B0533C0]<http://www.dickssportinggoods.com/home/index.jsp>

Todd Henry
Sr Technical Manager, Real Estate Market Research
345 Court Street  l  Coraopolis, PA  15108
todd.henry@dcsg.com<mailto:todd.henry@dcsg.com>  l  o: 724.273.4026  l  m: 412.522.3269

[cid:image002.png@01CEB48E.2B0533C0]<http://www.facebook.com/dickssportinggoods>[cid:image003.jpg@01CEB48E.2B0533C0]<https://twitter.com/DICKS>

**Comment by ecaldwell** [comment]
OK, I see it now. There's a `tables` object in the web map data that was not being handled in the app. I added this and will push the update to the hosted site later today. In the meantime, you can edit the JSON directly. Scroll past the `basemapLayers` entry in your web map's data and you should see an entry for `tables`.

Thanks for pointing this one out!

**Comment by henry296** [comment]
Evan,

Thanks for the fix.

I tried to edit the JSON by going to View an Items JSON and then click the pencil to edit the URL of those tables and then I clicked save, but my web map is still showing the old URL and when I go back into the View JSON task, it has the old service URL.  Is there another step that I need to do to edit the JSON?  I’m interested because there are some other opportunities as I migrate my maps and apps from one server to another that I need to change within the web app like the Search URLs.

Thanks.


[cid:image001.jpg@01CEB48E.2B0533C0]<http://www.dickssportinggoods.com/home/index.jsp>

Todd Henry
Sr Technical Manager, Real Estate Market Research
345 Court Street  l  Coraopolis, PA  15108
todd.henry@dcsg.com<mailto:todd.henry@dcsg.com>  l  o: 724.273.4026  l  m: 412.522.3269

[cid:image002.png@01CEB48E.2B0533C0]<http://www.facebook.com/dickssportinggoods>[cid:image003.jpg@01CEB48E.2B0533C0]<https://twitter.com/DICKS>
